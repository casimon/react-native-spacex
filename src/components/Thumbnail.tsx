import React from 'react';
import {Image, ImageStyle, View, StyleSheet, StyleProp} from 'react-native';
import images from '../images';

const styles = StyleSheet.create({
  image: {
    width: 50,
    height: 50
  }
});

interface Props {
  url?: string;
  style?: StyleProp<ImageStyle>;
}

class Thumbnail extends React.PureComponent<Props> {
  render() {
    const {url, style} = this.props;
    return (
      <View>
        {url ? (
          <Image
            style={[styles.image, style]}
            source={{
              uri: url
            }}
            resizeMode="contain"
          />
        ) : (
          <Image
            style={[styles.image, {borderRadius: 6}]}
            source={images.missing}
            resizeMode="contain"
          />
        )}
      </View>
    );
  }
}

export default Thumbnail;
