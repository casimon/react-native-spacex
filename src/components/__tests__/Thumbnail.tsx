/**
 * @format
 */

import 'react-native';
import React from 'react';
import Thumbnail from '../Thumbnail';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders thumbnail correctly with missing image', () => {
  const tree = renderer.create(<Thumbnail />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders thumbnail correctly with patch image', () => {
  const tree = renderer
    .create(<Thumbnail url="https://images2.imgbox.com/12/47/6uim8L1a_o.png" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders thumbnail correctly with patch image and option style prop', () => {
  const tree = renderer
    .create(
      <Thumbnail
        url="https://images2.imgbox.com/12/47/6uim8L1a_o.png"
        style={{width: 30, height: 30}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
