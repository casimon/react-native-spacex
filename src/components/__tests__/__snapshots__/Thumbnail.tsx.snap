// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders thumbnail correctly with missing image 1`] = `
<View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/images/missing.png",
      }
    }
    style={
      Array [
        Object {
          "height": 50,
          "width": 50,
        },
        Object {
          "borderRadius": 6,
        },
      ]
    }
  />
</View>
`;

exports[`renders thumbnail correctly with patch image 1`] = `
<View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://images2.imgbox.com/12/47/6uim8L1a_o.png",
      }
    }
    style={
      Array [
        Object {
          "height": 50,
          "width": 50,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`renders thumbnail correctly with patch image and option style prop 1`] = `
<View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://images2.imgbox.com/12/47/6uim8L1a_o.png",
      }
    }
    style={
      Array [
        Object {
          "height": 50,
          "width": 50,
        },
        Object {
          "height": 30,
          "width": 30,
        },
      ]
    }
  />
</View>
`;
